content/fundamentals/execution-context.md

```diff
@@ -40,30 +57,46 @@ switchToHttp(): HttpArgumentsHost;
 switchToWs(): WsArgumentsHost;
 ```
 
-Let's rewrite the previous example and use the `switchToHttp()` method now. The `host.switchToHttp()` helper call returns us an `HttpArgumentsHost` object. The `HttpArgumentsHost` object, in turn, has two useful methods. We use these methods to extract the desired objects, also using the Express type assertions in this case to return native Express typed objects:
+Let's rewrite the previous example using the `switchToHttp()` method. The `host.switchToHttp()` helper call returns an `HttpArgumentsHost` object that is appropriate for the HTTP application context. The `HttpArgumentsHost` object has two useful methods we can use to extract the desired objects. We also use the Express type assertions in this case to return native Express typed objects:
 
 ```typescript
 const ctx = host.switchToHttp();
 const request = ctx.getRequest<Request>();
 const response = ctx.getResponse<Response>();
 ```
 
-#### Current context
+Similarly, `RpcArgumentsHost` and `WsArgumentsHost` have methods to return appropriate objects in the microservices and WebSockets contexts.
 
-When building more generic [guards](/guards), [filters](/exception-filters), and [interceptors](/interceptors), we need a way to determine what's the type of application that our method is currently running for. This can be accomplished with the `getType()` method:
+```typescript
+export interface WsArgumentsHost {
+  /**
+   * Returns the data object.
+   */
+  getData<T = any>(): T;
+  /**
+   * Returns the client object.
+   */
+  getClient<T = any>(): T;
+}
+```
 
```

Comment:
Can we add a single sentence between these two examples? (it looks slightly better 😄 )